File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m - 1, j = n - 1, index = nums1.length - 1;

        while (i > -1 && j > -1) {
            if (nums2[j] > nums1[i]) {
                nums1[index--] = nums2[j--];
            } else {
                nums1[index--] = nums1[i--];
            }
        }

        while (j > -1) {
           nums1[index--] = nums2[j--];
        }


    }
}

File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
        int arr[n + m];
        int idx = 0;

        for (int i = 0; i < m; i++) {
            arr[i] = nums1[i];
        }

        for (int i = 0; i < n; i++) {
            arr[m + i] = nums2[i];
        }

        sort(arr, arr + (m + n));

        for (int i = 0; i < (n + m); i++) {
            nums1[i] = arr[i];
        }
    }
};